### PR TITLE
Feature/suggestion api

### PR DESCRIPTION
**Pull request by kand617**
Added new endpoint for suggesting keywords.  /suggestions
Added Service that periodically fetches the APIs in the database to rebuild the index.

### CODE REVIEW

File: src/main/java/io/apicollab/server/service/SuggestionService.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            //Build query
+            Query fuzzyQuery = new FuzzyQuery(new Term(INDEX_FIELD_NAME, partialKeyword));
+            StandardQueryParser queryParserHelper = new StandardQueryParser();
+            queryParserHelper.setAllowLeadingWildcard(true);
+            Query query = null;
+            try {
+                query = queryParserHelper.parse(wildCardKeyword, INDEX_FIELD_NAME);
+
+            } catch (QueryNodeException e) {
+                e.printStackTrace();
```
Comment (sudhirtumati) [inline src/main/java/io/apicollab/server/service/SuggestionService.java]: replace with `log.error`

File: src/main/java/io/apicollab/server/service/SuggestionService.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+     * Process each keyword by adding the search index
+     * @param suggestions
+     */
+    private void processSuggestions(Set<String> suggestions){
+
+        log.debug("Starting to index {} words", suggestions.size());
+        Analyzer analyzer = new StandardAnalyzer();
+        IndexWriterConfig config = new IndexWriterConfig(analyzer);
+        config.setOpenMode(IndexWriterConfig.OpenMode.CREATE);
+        this.directory = new RAMDirectory();
```
Comment (sudhirtumati) [inline src/main/java/io/apicollab/server/service/SuggestionService.java]: Create a new instance and update current working copy after indexing is complete

File: src/main/java/io/apicollab/server/service/SuggestionService.java (hunk 1 above)
Comment (kand617) [reply src/main/java/io/apicollab/server/service/SuggestionService.java]: done

File: src/main/java/io/apicollab/server/service/SuggestionService.java (hunk 2 above)
Comment (kand617) [reply src/main/java/io/apicollab/server/service/SuggestionService.java]: done

**Pull request merged**